src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionAzureKeyVaultProvider.java

```diff
@@ -49,668 +53,682 @@
-        int currentIndex = firstVersion.length;
-        short keyPathLength = convertTwoBytesToShort(encryptedColumnEncryptionKey, currentIndex);
-        // We just read 2 bytes
-        currentIndex += 2;
-
-        // Get ciphertext length
-        short cipherTextLength = convertTwoBytesToShort(encryptedColumnEncryptionKey, currentIndex);
-        currentIndex += 2;
-
-        // Skip KeyPath
-        // KeyPath exists only for troubleshooting purposes and doesnt need validation.
-        currentIndex += keyPathLength;
-
-        // validate the ciphertext length
-        if (cipherTextLength != keySizeInBytes) {
-            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_AKVKeyLengthError"));
-            Object[] msgArgs = {cipherTextLength, keySizeInBytes, masterKeyPath};
-            throw new SQLServerException(this, form.format(msgArgs), null, 0, false);
+        /**
+         * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider to authenticate to AAD. This is used by
+         * KeyVaultClient at runtime to authenticate to Azure Key Vault.
+         */
+        SQLServerColumnEncryptionAzureKeyVaultProvider() throws SQLServerException {
+                createKeyvaultClients(new ManagedIdentityCredentialBuilder().build());
         }
 
-        // Validate the signature length
-        int signatureLength = encryptedColumnEncryptionKey.length - currentIndex - cipherTextLength;
+        /**
+         * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider to authenticate to AAD. This is used by
+         * KeyVaultClient at runtime to authenticate to Azure Key Vault.
+         *
+         * @param clientId Identifier of the client requesting the token.
+         */
+        SQLServerColumnEncryptionAzureKeyVaultProvider(String clientId) throws SQLServerException {
+                if (null == clientId || clientId.isEmpty()) {
+                        MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_NullValue"));
+                        Object[] msgArgs1 = {"Client ID"};
+                        throw new SQLServerException(form.format(msgArgs1), null);
+                }
+                createKeyvaultClients(new ManagedIdentityCredentialBuilder().clientId(clientId).build());
+        }
 
-        if (signatureLength != keySizeInBytes) {
-            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_AKVSignatureLengthError"));
-            Object[] msgArgs = {signatureLength, keySizeInBytes, masterKeyPath};
-            throw new SQLServerException(this, form.format(msgArgs), null, 0, false);
+        /**
+         * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider using the provided TokenCredential to
+         * authenticate to AAD. This is used by KeyVaultClient at runtime to authenticate to Azure Key Vault.
+         *
+         * @param tokenCredential The TokenCredential to use to authenticate to Azure Key Vault.
+         */
```

Comment:
[WARNING] Javadoc Warnings
[WARNING] C:\mssql-jdbc\src\main\java\com\microsoft\sqlserver\jdbc\SQLServerColumnEncryptionAzureKeyVaultProvider.java:145: warning: no @throws for com.microsoft.sqlserver.jdbc.SQLServerException
[WARNING] public SQLServerColumnEncryptionAzureKeyVaultProvider(TokenCredential tokenCredential)
[WARNING] ^